test/rubygems/test_gem_package.rb

```diff
@@ -124,12 +124,10 @@ def test_build_time_source_date_epoch
     ENV["SOURCE_DATE_EPOCH"] = epoch
   end
 
-  def test_build_time_source_date_epoch_automatically_set
+  def test_build_time_without_source_date_epoch
```

Comment:
The reason I changed it is because the old one assumed that the default value is set between `start_time = ...` and `end_time = ...`. It being set at-or-after `start_time` is not guaranteed anymore and was never an actual requirement. In retrospect, I think that assertion was accidentally testing an implementation detail.

However, looking at it again, the `end_time = ...` + associated assertion should be able to be re-added without issue. Does adding those two lines back work for you, @bronzdoc? If not, lmk what you'd prefer.

Reply:
Ah I see. I think is ok as it is @duckinator